Abstracts adding <meta> tags to the output

Sesquipedalian opened the pull request:
~~The property and content values for the Open Graph <meta> tags are stored as key/value pairs in $context['open_graph']. Mods can add/overwrite OG tags either by calling `addOpenGraph()` directly, or by calling `call_integration_hook('integrate_theme_context')` and then editing the content of $context['open_graph'] themselves.~~

Moves all content related `<meta>` tag info into `$context['meta_tags']`. This will allow mods to add/edit/remove arbitrary meta info (e.g. Open Graph tags) simply by accessing `$context['meta_tags']`—ideally in a function that hooks into the `integrate_theme_context` hook, but theoretically anywhere.

Sesquipedalian committed:
Abstracts adding Open Graph tags to the output
The property and content values for the Open Graph <meta> tags are stored as key/value pairs in $context['open_graph']. Mods can add/overwrite OG tags either by calling `addOpenGraph()` directly, or by calling `call_integration_hook('integrate_theme_context')` and then editing the content of $context['open_graph'] themselves.

Signed-off-by: Jon Stovell <jonstovell@gmail.com>

jdarwood007 commented:
With how basic the function is, maybe we could just rename it addMetaTag or similar and cleanup all meta data in the theme.  Because as it is, anything can call the function to add meta data.

Sesquipedalian commented:
Not a bad idea. I'll need to adjust the input parameters a bit in order to allow setting arbitrary properties, but that should be trivial.

Sesquipedalian commented:
On further consideration, the function is unnecessary. We can just add meta info to $context directly.

Sesquipedalian committed:
Abstracts to handle any arbitrary <meta> attributes
Gets rid of unnecessary function. Why not just add stuff to $context['meta_tags'] directly?

Signed-off-by: Jon Stovell <jonstovell@gmail.com>

XinYenFon commented:
I think its good to go.

Sesquipedalian committed:
Define $context['meta_tags'] as empty array early in SMF's process

Signed-off-by: Jon Stovell <jonstovell@gmail.com>

Sesquipedalian commented:
> Can you define $context['meta_tags'] as an empty array first?  Ideally and preferable very early in SMF's process

Done as requested, @MissAllSunday.
